html/user/openssl_sign_form.php

```diff
@@ -0,0 +1,58 @@
+<?php
+// This file is part of BOINC.
+// http://boinc.berkeley.edu
+// Copyright (C) 2008 University of California
+//
+// BOINC is free software; you can redistribute it and/or modify it
+// under the terms of the GNU Lesser General Public License
+// as published by the Free Software Foundation,
+// either version 3 of the License, or (at your option) any later version.
+//
+// BOINC is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+// See the GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with BOINC.  If not, see <http://www.gnu.org/licenses/>.
+
+require_once("../inc/boinc_db.inc");
+require_once("../inc/user.inc");
+require_once("../inc/util.inc");
+require_once("../inc/countries.inc");
+require_once('../inc/recaptchalib.php');
+
+check_get_args(array("tnow", "ttok"));
+
+$user = get_logged_in_user();
+check_tokens($user->authenticator);
+
+page_head(tra("Generate UserID ownership signature"), null, null, null, boinc_recaptcha_get_head_extra());
+
+// Verify the user is logged in
+if ($user) {
+    // The user is logged in
```

Comment:
We check for the public/private keys existence in the [action file](https://github.com/BOINC/boinc/pull/2134/files/29d3e1a612aae10c6a88e38f4526bef76fd5b23e#diff-91d6b04155bcf59e677280a695167c2fR68) however the error messages are generic/informative instead of instructing the user to bother the admin. So you'd prefer the check to be performed in the form rather than in the action?

Reply:
If the user sees the form and the button to generate the signature is enabled that suggests that the system is working. Getting a generic error message after entering the data copied from an external site leads to frustration on the users side.